server/boot/user.js

```diff
@@ -223,9 +229,20 @@ module.exports = function(app) {
             +challenge.challengeType === 4;
         }
 
-        const completedChallenges = profileUser.completedChallenges.filter(
-          ({ name }) => typeof name === 'string'
-        );
+        const completedChallenges = profileUser.completedChallenges
+          .filter(({ name }) => typeof name === 'string')
+          .map(challenge => {
+              const format = 'MMM DD, YYYY';
+              if (challenge.completedDate) {
```

Comment:
This constant should be hoisted to the top of the file with the rest of the constants and given a specific name. While normally not important, we serve 10's of thousands of people a day and growing. Minor memory tricks like this helps us reduce server load.